staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go

```diff
@@ -1880,3 +1880,72 @@ func (l *fakeNodeLister) Get(name string) (*v1.Node, error) {
 	}
 	return nil, nil
 }
+
+func TestLoadBalancerClass(t *testing.T) {
+	testcases := []struct {
+		name            string
+		service         *v1.Service
+		expectOp        loadBalancerOperation
+		expectCloudCall bool
+	}{
+		{
+			name: "create a load balancer type of service with loadBalancerClass",
+			service: &v1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-load-balancer-class",
+					Namespace: "default",
+				},
+				Spec: v1.ServiceSpec{
+					Type: v1.ServiceTypeLoadBalancer,
+					Ports: []v1.ServicePort{{
+						Port: int32(80),
+					}},
+					LoadBalancerClass: utilpointer.StringPtr("test.com/test"),
+				},
+			},
+			expectOp:        deleteLoadBalancer,
+			expectCloudCall: false,
+		},
+		{
+			name: "create a load balancer type of service without LoadBalancerClass",
+			service: &v1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-load-balancer-class",
+					Namespace: "default",
+				},
+				Spec: v1.ServiceSpec{
+					Type: v1.ServiceTypeLoadBalancer,
+					Ports: []v1.ServicePort{{
+						Port: int32(80),
+					}},
+				},
+			},
+			expectOp:        ensureLoadBalancer,
+			expectCloudCall: true,
+		},
+	}
+	for _, testcase := range testcases {
+		t.Run(testcase.name, func(t *testing.T) {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+			controller, cloud, client := newController()
+			key := fmt.Sprintf("%s/%s", testcase.service.Namespace, testcase.service.Name)
+			if _, err := client.CoreV1().Services(testcase.service.Namespace).Create(ctx, testcase.service, metav1.CreateOptions{}); err != nil {
+				t.Fatalf("Failed to prepare service %s for testing: %v", key, err)
+			}
+			op, err := controller.syncLoadBalancerIfNeeded(ctx, testcase.service, key)
```

Comment:
How similar are these tests to https://github.com/kubernetes/kubernetes/blob/0fcd225acbd428d22d7708cb1d97ca131d57ddc4/staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go#L223-L260?

Reply:
That should be the same, I will remove those two cases tonight, thanks